### PR TITLE
Adds a new base assault variant that places the URF in an attacking role against the unsc

### PR DESCRIPTION
**Pull request by X0-11**
:cl: XO-11
rscadd: Modifies base assault to allow vehicles to be obtained from ONI base.
rscadd: A new base assault variant is available in the mapswitch vote list. The urf attacks the UNSC moonbase.
tweak: Modifies a wide range of vehicle stats to make them more responsive and less painful to utilise.
rscadd: Buffs the SMG by transferring the slowdown removal that pistols get to it.
tweak: Targeting destructible objects via direct click now guarantees a hit on them.
tweak: Makes mobile respawn have a downside to regular respawn, respawn time. Mobile respawn takes 8 minutes instead of 5.
/:cl:

### CODE REVIEW

**Comment by enkidienne** [comment]
this looks like a lot of stuff in a single PR. touching a lot of stuff that isn't necessarily related to the gamemode?

**Comment by bloxgate** [comment]
Yes, this seems to contain commits from a previous PR. Can you update your local branch to match our current alpha branch?

**Comment by X0-11** [comment]
Don't seem to be able to clean it up. I can make sure I update my alpha branch for later PRs though.

**Comment by X0-11** [comment]
ignore that, i just hadn't recieved an updated view so it still looked like it had the previous PR things as commits

**Review by BDpuffy420 — APPROVED** [review]

File: code/modules/halo/weapons/M6D.dm [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 /obj/item/weapon/gun/projectile/m6d_magnum
 	name = "\improper M6D Magnum"
-	desc = "A common UNSC sidearm and one of the variants of Misriah Armory's M6 handgun series. Takes 12.7mm calibre magazines."
+	desc = "A common UNSC sidearm and one of the variants of Misriah Armory's M6 handgun series. Takes 12.7mm calibre magazines. The relatively small size and low weight makes it unencumbering and allows for faster movement than when utilising full rifles"
```
Comment (Shadowtail117) [inline code/modules/halo/weapons/M6D.dm]:
```suggestion
	desc = "A common UNSC sidearm and one of the variants of Misriah Armory's M6 handgun series. Takes 12.7mm calibre magazines. Its small size and weight make it unencumbering, allowing for faster movement than when utilising fully-sized rifles."
```

File: code/modules/halo/weapons/M6S.dm [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 /obj/item/weapon/gun/projectile/m6c_magnum_s
 	name = "\improper M6S silenced magnum"
-	desc = "The M6C/SOCOM is a special operations variant of the popular M6C but with a whole host of inbuilt attachments. Takes 12.7mm calibre magazines."
+	desc = "The M6C/SOCOM is a special operations variant of the popular M6C but with a whole host of inbuilt attachments. Takes 12.7mm calibre magazines.The relatively small size and low weight makes it unencumbering and allows for faster movement than when utilising full rifles"
```
Comment (Shadowtail117) [inline code/modules/halo/weapons/M6S.dm]:
```suggestion
	desc = "The M6C/SOCOM is a special operations variant of the popular M6C but with a whole host of inbuilt attachments. Takes 12.7mm calibre magazines. Its small size and weight make it unencumbering, allowing for faster movement than when utilising fully-sized rifles."
```

File: code/modules/halo/weapons/M7.dm [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 /obj/item/weapon/gun/projectile/m7_smg
 	name = "M7 submachine gun"
-	desc = "The M7/Caseless Submachine Gun is a fully automatic close quarters infantry and special operations weapon. Takes 5mm calibre magazines."
+	desc = "The M7/Caseless Submachine Gun is a fully automatic close quarters infantry and special operations weapon. Takes 5mm calibre magazines.The relatively small size and low weight makes it unencumbering and allows for faster movement than when utilising full rifles"
```
Comment (Shadowtail117) [inline code/modules/halo/weapons/M7.dm]:
```suggestion
	desc = "The M7/Caseless Submachine Gun is a fully automatic close quarters infantry and special operations weapon. Takes 5mm calibre magazines. Its small size and weight make it unencumbering, allowing for faster movement than when utilising fully-sized rifles."
```

File: code/modules/halo/weapons/covenant/pistols.dm [hunk 4]
```diff
@@ -1,7 +1,7 @@
 
 /obj/item/weapon/gun/energy/plasmapistol
 	name = "Type-25 Directed Energy Pistol"
-	desc = "A dual funtionality pistol: It fires bolts of plasma, and when overcharged is capable of emitting a small emp burst at the point of impact."
+	desc = "A dual funtionality pistol: It fires bolts of plasma, and when overcharged is capable of emitting a small emp burst at the point of impact.The relatively small size and low weight makes it unencumbering and allows for faster movement than when utilising full rifles"
```
Comment (Shadowtail117) [inline code/modules/halo/weapons/covenant/pistols.dm]:
```suggestion
	desc = "A dual-functionality pistol: It fires bolts of plasma, and when overcharged is capable of emitting a small emp burst at the point of impact. Its small size and weight make it unencumbering, allowing for faster movement than when utilising fully-sized rifles."
```

File: code/modules/halo/weapons/covenant/pistols.dm [hunk 5]
```diff
@@ -122,7 +122,7 @@
 
 /obj/item/weapon/gun/projectile/needler // Uses "magazines" to reload rather than inbuilt cells.
 	name = "Type-33 Guided Munitions Launcher"
-	desc = "This weapon fire razor-sharp crystalline shards which can explode violently when embedded into targets."
+	desc = "This weapon fire razor-sharp crystalline shards which can explode violently when embedded into targets.The relatively small size and low weight makes it unencumbering and allows for faster movement than when utilising full rifles"
```
Comment (Shadowtail117) [inline code/modules/halo/weapons/covenant/pistols.dm]:
```suggestion
	desc = "This weapon fires razor-sharp crystalline shards which can explode violently when embedded into targets. Its small size and weight make it unencumbering, allowing for faster movement than when utilising fully-sized rifles."
```

File: maps/base_assault_unsc_defend_vs_urf/urf_unsc_gm.dm [hunk 6]
```diff
@@ -0,0 +1,13 @@
+
+/datum/game_mode/base_assault/unsc_urf
+	name = "Base Assault (UNSC vs URF)"
+	config_tag = "base_assault_unsc_urf"
+	defenders = "The UNSC"
+	attackers = " The URF"
```
Comment (Shadowtail117) [inline maps/base_assault_unsc_defend_vs_urf/urf_unsc_gm.dm:6]:
```suggestion
	attackers = "The URF"
```

**Review by Shadowtail117 — CHANGES_REQUESTED** [review]
Some grammatical fixes. As an aside, given this PR touches a lot more than simply URF base assault consider changing the title.

**Pull request merged**